### PR TITLE
Fixed several bugs with notifications

### PR DESCRIPTION
**Pull request by ekzobrain**
Fixed method JsonRpcConsumer.notify_channel()
Fixed inbound http/websocket notifications to return correct transport-specific results

### CODE REVIEW

File: channels_jsonrpc/jsonrpcconsumer.py [hunk 1]
```diff
@@ -253,12 +267,12 @@ def __handle(self, content, message):
             else:
                 if isinstance(data, dict):
 
-                    if data.get('method') is not None and data.get('params') is not None and data.get('id') is None:
```
Comment (millerf) [inline channels_jsonrpc/jsonrpcconsumer.py:256]: @soulhunter1987 
Why did you remove the check for 'params' and 'method'? 
That is what make the tests fail.

If sending {"value": "my_value"} which should be received as a wrongly formatted  JSON-RPC frame, it handles it as a notification,....

File: channels_jsonrpc/jsonrpcconsumer.py [hunk 2]
```diff
@@ -231,10 +243,11 @@ def raw_receive(self, message, **kwargs):
         :return:
         """
         content = '' if "text" not in message else message["text"]
-        result = self.__handle(content, message)
+        result, result_type = self.__handle(content, message)
 
-        # Send responce back
-        self.send(text=self._encode(result))
+        # Send responce back only if it is a call, not notification
+        if result_type == 0:
```
Comment (millerf) [inline channels_jsonrpc/jsonrpcconsumer.py]: @soulhunter1987 
please use some GLOBAL_VAR for the result_type:
RESULT_NOTIFICATION_TYPE/RESULT_REQUEST_TYPE, for clarity purposes, or 'is_notification as variable name' ...

**Review by millerf — CHANGES_REQUESTED** [review]

File: channels_jsonrpc/jsonrpcconsumer.py (hunk 2 above)
Comment (ekzobrain) [reply channels_jsonrpc/jsonrpcconsumer.py]: Of course it should be done, but for now It is just a quick working solution. These lines would anywhere look different after moving to a JSON-RPC decoding/encoding library, and I plan to do it soon, because I need support of batch calls, and, may be, protocol versions 1.1/1.2 along with 2.0. So this is a temporary working solution.

File: channels_jsonrpc/jsonrpcconsumer.py (hunk 1 above)
Comment (ekzobrain) [reply channels_jsonrpc/jsonrpcconsumer.py:256]: You are right here, I did not think about such case. I'll change this back now...

Comment (ekzobrain) [reply channels_jsonrpc/jsonrpcconsumer.py:256]: Done.

File: channels_jsonrpc/jsonrpcconsumer.py (hunk 2 above)
Comment (ekzobrain) [reply channels_jsonrpc/jsonrpcconsumer.py]: Pushed a quick fix

**Comment by ekzobrain** [comment]
I pushed one more fix with contant usage of JsonRpcConsumer._encode() to simplify json encoder configuration.
